Framework/DataHandling/src/LoadMuonNexusV2Helper.cpp

```diff
@@ -188,6 +203,66 @@ getLoadedDetectors(const DataObjects::Workspace2D_sptr &localWorkspace) {
   return loadedDetectors;
 }
 
+MuonNexus::SampleInformation
+loadSampleInformationFromNexus(const NeXus::NXEntry &entry) {
+  auto runSample = entry.openNXGroup(NeXusEntry::SAMPLE);
+  MuonNexus::SampleInformation sampleInformation;
+  try {
+    sampleInformation.magneticField =
+        runSample.getFloat(NeXusEntry::MAGNETICFIELD);
+    sampleInformation.temperature = runSample.getFloat(NeXusEntry::TEMPERATURE);
+  } catch (std::runtime_error) {
+    throw std::runtime_error("Could not load sample information (temperature "
+                             "and magnetic field) from nexus entry");
+  }
+  return sampleInformation;
+}
+/**
+ * Loads default detector grouping, if this isn't present
+ * return dummy grouping
+ * @returns :: Grouping table
+ */
+Workspace_sptr loadDefaultDetectorGrouping(
+    const DataObjects::Workspace2D_sptr &localWorkspace) {
+
+  auto instrument = localWorkspace->getInstrument();
+  auto &run = localWorkspace->run();
+  std::string mainFieldDirection =
+      run.getLogData("main_field_direction")->value();
+  API::GroupingLoader groupLoader(instrument, mainFieldDirection);
+  try {
+    const auto idfGrouping = groupLoader.getGroupingFromIDF();
+    return idfGrouping->toTable();
+  } catch (const std::runtime_error &) {
+    auto dummyGrouping = std::make_shared<Grouping>();
+    if (instrument->getNumberDetectors() != 0) {
+      dummyGrouping = groupLoader.getDummyGrouping();
+    } else {
+      // Make sure it uses the right number of detectors
+      std::ostringstream oss;
+      oss << "1-" << localWorkspace->getNumberHistograms();
```

Comment:
Rather than create a stringstream (which is fairly heavy) for a single concat operation you can just do

`std::string numDetectors = "1-" + std::to_string(localWorkspace...);`
`dummyGrouping->groups.emplace_back(std::move(numDetectors));`